python/lsst/meas/algorithms/astrometrySourceSelector.py

```diff
@@ -122,12 +124,16 @@ def _hasCentroid_vector(self, sourceCat):
         """Return True for each source that has a valid centroid"""
         return np.isfinite(sourceCat.get(self.centroidXKey)) \
             & np.isfinite(sourceCat.get(self.centroidYKey)) \
+            & np.isfinite(sourceCat.get(self.centroidXSigmaKey)) \
+            & np.isfinite(sourceCat.get(self.centroidYSigmaKey)) \
             & ~sourceCat.get(self.centroidFlagKey)
 
     def _hasCentroid(self, source):
         """Return True if the source has a valid centroid"""
         centroid = source.getCentroid()
-        return np.all(np.isfinite(centroid)) and not source.getCentroidFlag()
+        centroidErr = source.getCentroidErr()
+        return np.all(np.isfinite(centroid)) and np.all(np.isfinite(centroidErr)) \
+            and not source.getCentroidFlag()
```

Comment:
`np.all(np.isfinite(centroid))` should always be unnecessary.  If you really want to check, you could `assert` on it - it should be finite even when `source.getCentroidFlag()` is true.

Reply:
Ok, I guess I can turn the centroid finiteness checks into asserts, but I'm going to argue in an RFC shortly that the errors should also be finite for unflagged sources.